sway/input/seat.c

```diff
@@ -652,19 +659,29 @@ void seat_set_focus_warp(struct sway_seat *seat,
 
 		if (last_focus) {
 			seat_send_unfocus(last_focus, seat);
+			container_set_dirty(last_focus);
 		}
 		seat_send_focus(container, seat);
 
 		container_set_dirty(container);
 		container_set_dirty(container->parent); // for focused_inactive_child
-		if (last_focus) {
-			container_set_dirty(last_focus);
-		}
+	}
+
+	// emit ipc events
+	if (notify && new_workspace && last_workspace != new_workspace) {
```

Comment:
Somehow, no. I haven't tested it fully, but sometimes new_workspace is NULL. I know because I tried to replace https://github.com/swaywm/sway/blob/d6a5dfc913ba958d38af907ddb7acd7483a1cee7/sway/input/seat.c#L619 with  `new_output = new_workspace` to optimise it and it would crash because `new_workspace` is NULL, even though new_output ends up not being NULL. Very confusing.

Reply:
It occurs when moving a workspace to a different output, since it focuses the new output, so `new_workspace` is NULL but `new_output` is not.